data-reporting-service/observation-service/src/test/java/gov/cdc/etldatapipeline/observation/service/ObservationServiceTest.java

```diff
@@ -1,76 +1,116 @@
-//
-//        // Test the method
-//        observationService.processMessage(streamsBuilder);
-//
-//        // Verify that expected methods were called
-//        verify(streamsBuilder).stream(eq(observationService.observationTopicOutput), eq(consumed));
-//        verify(observationRepository).computeObservations(eq(observationUid));
-//    }
-//}
+package gov.cdc.etldatapipeline.observation.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import gov.cdc.etldatapipeline.observation.TestUtils;
+import gov.cdc.etldatapipeline.observation.repository.IObservationRepository;
+import gov.cdc.etldatapipeline.observation.repository.model.Observation;
+import gov.cdc.etldatapipeline.observation.util.ProcessObservationDataUtil;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.*;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.kafka.core.KafkaTemplate;
+
+import java.util.Optional;
+import java.util.Properties;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+
+class ObservationServiceTest {
+
+    @Mock
+    private IObservationRepository observationRepository;
+
+    @Mock KafkaTemplate<String, String> kafkaTemplate;
+
+    @Captor
+    private ArgumentCaptor<String> topicCaptor;
+
+    @Captor
+    private ArgumentCaptor<String> messageCaptor;
+
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+    ProcessObservationDataUtil transformer = new ProcessObservationDataUtil();
+
+    @Test
+    void testProcessMessage() {
+        String observationTopic = "Observation";
+        String observationTopicOutput = "ObservationOutput";
+        String observationTopicOutputTransformed = "ObservationOutputTransformed";
+
+        // Mocked input data
+        Long observationUid = 123456789L;
+        String obsDomainCdSt = "Order";
+        String payload = "{\"payload\": {\"after\": {\"observation_uid\": \"" + observationUid + "\"}}}";
+
+        Observation observation = constructObservation(observationUid, obsDomainCdSt);
+        when(observationRepository.computeObservations(eq(String.valueOf(observationUid)))).thenReturn(Optional.of(observation));
+
+        final String expectedTransformed = transformer.transformObservationData(observation).toString();
+
+        validateData(observationTopic, observationTopicOutput,
+                observationTopicOutputTransformed, payload, observationUid);
+
+        verify(kafkaTemplate).send(topicCaptor.capture(), messageCaptor.capture());
```

Comment:
The most interesting feature discovered in Mockito in the contest of this testing 🧐